pkg/parquetquery/iters.go

```diff
@@ -964,16 +968,17 @@ func (c *SyncIterator) closeCurrRowGroup() {
 }
 
 func (c *SyncIterator) makeResult(t RowNumber, v *pq.Value) *IteratorResult {
-	r := DefaultPool.Get()
-	r.RowNumber = t
-	if c.selectAs != "" {
+	// Use same static result instead of pooling
+	c.at.RowNumber = t
+	if len(c.at.Entries) == 1 {
```

Comment:
i'm missing how the "selectAs" logic in NewSyncIterator() interacts with this code here. 

Can Entries only have one value? should we make it a single value and not a slice?

Reply:
For a SyncIterator which reads a column directly, it will always have 1 value.  But join-type iterators will have multiple, for example at the span-level both start time and name.   Because we can link iters in any combination, a join on top of raw iters and child join iters, Entries is a slice that works for every type.  It would be complex (but maybe not impossible) to make SyncIterators return something with `Entry struct{}` and join-type iterators with `Entries []struct{}`.

When `c.selectAs` is empty string, it means we just need the row number, so `Entries` is empty.  Checking the slice len is just a little faster than doing the string comparison, and there's no need to resize the slice every iteration. We know during creation if the SyncIterator will have 0 or 1 value.

Reply:
understood. perhaps add a comment about the relationship between selectAs and the entries? but lgtm!

approval stands